server/middleware/Authenticate.js

```diff
@@ -0,0 +1,585 @@
+import dotenv from 'dotenv';
+import jwt from 'jsonwebtoken';
+import db from '../models';
+import Helper from '../Helper/utility';
+
+dotenv.config();
+const secretKey = process.env.SECRET;
+
+// const Roles = db.Roles;
+/**
+ * to authenticate users
+ */
+const authenticate = {
+  /**
+   * verifyToken - Verifies if a token supplied is valid or not
+   *
+   * @param  {Object} request  Request Object
+   * @param  {Object} response  Response Object
+   * @param  {Object} next Middleware function
+   * @returns {Object} Response Status
+   */
+  verifyToken(request, response, next) {
+    // token request
+    const token = request.body.token ||
+      request.get('Authorization') ||
+      request.headers['x-access-token'];
+    if (token) {
+        // If the token is valid, request is set to decoded
+      jwt.verify(token, secretKey, (error, decoded) => {
+        if (error) {
+          response.status(401).send({
+            message: 'Invalid token, Authentication required'
+          });
+        }
+          // decode token with user id
+        request.decoded = decoded;
+          // Move to the next middleware function
+        next();
+      });
+    } else {
+      response.status(401).send({
+        message: 'You are not permitted to perform this action'
```

Comment:
your response messages should be descriptive enough for the user to understand why that operation can not be performed.